src/macaroni/model/element/Pipe.java

```diff
@@ -108,8 +108,15 @@ public ActiveElement getEndpoint(int index) {
      */
     @Override
     public boolean enter(Character character, Element from) {
-        Random.setCharacterLocationBeforeEnteringPipe(this, from);
-        if (neighbours.contains(from) && !occupied && endpoints.size() == 2) {
+        if (!occupied && endpoints.size() == 2) {
+            if (from == null) {
+                Random.setCharacterLocationBeforeEnteringPipe(this, endpoints.get(0));
+            } else if (neighbours.contains(from)) {
+                Random.setCharacterLocationBeforeEnteringPipe(this, from);
+            } else {
+                return false;
+            }
+
             if (effect.enter(character)) {
```

Comment:
I believe the effect should only be evaluated when 'from' is not null, because when spawning the players, we want the player to stay on the pipe that was specified as their starting location.

Reply:
Resolved in #20